lua/neocodeium/commands.lua

```diff
@@ -181,10 +185,12 @@ function M.restart()
 end
 
 function M.toggle()
-   vim.g.neocodeium_enabled = vim.g.neocodeium_enabled == false
-   if vim.g.neocodeium_enabled and not server.pid then
-      server:run()
-   end
+   M[vim.g.neocodeium_enabled and "disable" or "enable"]()
```

Comment:
This is hard no from me. 

1. I don't like "smart" code. I don't care if it would span few more lines, but this is unreadable, doesn't allow to jump to definition of those functions without manually searching.
2. It doesn't act like the old code, yes in the current plugin state it doesn't matter because I have added that g.neocodeium_enabled is always present and without user manually deletes it all would be fine, but still on missing variable this code behaves wrong. 

But at the same time you are right that this is a code duplication, so it should be addressed. 

I like to name variables with the on/off meaning as non-negative (bad English not sure how to explain this right), so I have named them enabled in the plugin. But maybe because they are not documented and you can say hidden from the user, it is better name them with negative meaning as vim.g and vim.b `neocodeium_disabled`? What is your opinion. 


TL;DR this, and especially buffer toggle should be rewritten using if statement.

Reply:
Awesome, thanks for the feedback! I revised the commits to use the check against false and moved them to `if`/`else` statements